src/FakeItEasy.Analyzer/ArgumentConstraintNullabilityMismatchCodeFixProvider.cs

```diff
@@ -0,0 +1,144 @@
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+#if CSHARP
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+#elif VISUAL_BASIC
+    using Microsoft.CodeAnalysis.VisualBasic;
+    using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+#endif
+
+#if CSHARP
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+#elif VISUAL_BASIC
+    [ExportCodeFixProvider(LanguageNames.VisualBasic)]
+#endif
+    public class ArgumentConstraintNullabilityMismatchCodeFixProvider : CodeFixProvider
+    {
+        private static readonly Task CompletedTask = Task.FromResult(false);
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(DiagnosticDefinitions.ArgumentConstraintNullabilityMismatch.Id);
+
+        private static string MakeConstraintNullableCodeFixTitle =>
+            DiagnosticDefinitions.ResourceManager.GetString(nameof(MakeConstraintNullableCodeFixTitle));
+
+        private static string MakeNotNullConstraintCodeFixTitle =>
+            DiagnosticDefinitions.ResourceManager.GetString(nameof(MakeNotNullConstraintCodeFixTitle));
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.FirstOrDefault();
+            if (diagnostic == null)
+            {
+                return CompletedTask;
+            }
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    MakeConstraintNullableCodeFixTitle,
+                    ct => this.MakeConstraintNullableAsync(context, diagnostic, ct)),
+                diagnostic);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    MakeNotNullConstraintCodeFixTitle,
+                    ct => this.MakeNotNullConstraintAsync(context, diagnostic, ct)),
+                diagnostic);
+
+            return CompletedTask;
+        }
+
+        private static MemberAccessExpressionSyntax GetConstraintNode(Diagnostic diagnostic, SyntaxNode root)
+        {
+            // getInnermostNodeForTie: true to disambiguate between ArgumentSyntax and
+            // MemberAccessExpressionSyntax, which have the same source span. We want the
+            // MemberAccessExpressionSyntax node, which is the innermost one.
+            return root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true) as MemberAccessExpressionSyntax;
+        }
+
+        private static MemberAccessExpressionSyntax SimpleMemberAccess(ExpressionSyntax expression, SimpleNameSyntax name)
+        {
+#if CSHARP
+            return SyntaxFactory.MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, expression, name);
+#elif VISUAL_BASIC
+            return SyntaxFactory.SimpleMemberAccessExpression(expression, name);
+#endif
+        }

```

Comment:
I don't know why the Roslyn API is different for C# and VB.NET in this case... there's no good reason IMO.